cassandra-oss/Dockerfile.ubi7

```diff
@@ -0,0 +1,183 @@
+# Copyright DataStax, Inc, 2020
+#   Please review the included LICENSE file for more information.
+#
+ARG BASETAG=7.8
+FROM registry.access.redhat.com/ubi7/ubi-minimal:${BASETAG} AS builder
+
+ARG URL_PREFIX=https://dlcdn.apache.org/cassandra
+ARG CASSANDRA_VERSION=trunk
+ARG METRICS_COLLECTOR_VERSION=0.3.4
+ARG TINI_VERSION=0.19.0
+
+ENV LANG='en_US.UTF-8' LANGUAGE='en_US:en' LC_ALL='en_US.UTF-8'
+
+# Update base layer
+RUN microdnf update && rm -rf /var/cache/yum
+
+# Install packages needed during install process
+RUN microdnf install shadow-utils tar gzip unzip && microdnf clean all
+
+# COPY in files if they are available locally
+COPY * /
+WORKDIR /
+
+# Download and extract Metrics Collector
+ENV MCAC_PATH /opt/metrics-collector
+RUN mkdir ${MCAC_PATH} && \
+    if test ! -e datastax-mcac-agent-0.1.9.tar.gz; then curl -L -O "https://github.com/datastax/metric-collector-for-apache-cassandra/releases/download/v${METRICS_COLLECTOR_VERSION}/datastax-mcac-agent-${METRICS_COLLECTOR_VERSION}.tar.gz"; fi && \
+    tar --directory ${MCAC_PATH} --strip-components 1 --gzip --extract --file datastax-mcac-agent-${METRICS_COLLECTOR_VERSION}.tar.gz
+
+# Install tini
+RUN if test ! -e tini-amd64; then curl -L -O "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini-amd64"; fi && \
+    mv tini-amd64 /usr/local/bin/tini && \
+    chmod 0755 /usr/local/bin/tini
+
+#############################################################
+
+# Build and extract Cassandra
+FROM --platform=$BUILDPLATFORM azul/zulu-openjdk-debian:8u345 as cass-builder
+ENV CASSANDRA_PATH /opt/cassandra
+WORKDIR /build
+RUN set -x \
+    && rm -fr /etc/apt/sources.list.d/* \
+    && rm -rf /var/lib/apt/lists/* \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends git ant ant-optional make maven \
+    && git clone https://github.com/apache/cassandra.git \
+    && cd cassandra \
+    && ant artifacts mvn-install \
+    && mkdir ${CASSANDRA_PATH} \
+    && tar --directory ${CASSANDRA_PATH} --strip-components 1 --gzip --extract --file /build/cassandra/build/apache-cassandra-4.2-SNAPSHOT-bin.tar.gz \
```

Comment:
the tar.gz bundle name is hard-coded, because that's what it happens to be today. Probably need to update this to dynamically figure it out based on what is built, possibly with a smart `find` command